### ISSUE TITLE
can't upload a plugin created by plugin-bootstrap

### ISSUE THREAD

**Issue by clofresh** [sjrd218/rundeck-verb-project#3]
I get a 500 error when I build a plugin from a template generated from [plugin-bootstrap](https://github.com/rundeck/plugin-bootstrap). The stack trace in the server logs is at the end of this issue.

Looks like it uses the [gradle jar task](https://github.com/rundeck-plugins/build-zip/blob/master/build.gradle#L5) to zip it up which by default includes the version number in the directory name, whereas the plugin repo api expects the zipped directory to just be the plugin name. If I manually zip up a plugin with just the plugin name as the directory, it works.

```
rundeck_1                   | 2018-10-12 23:07:16.224 ERROR --- [tp1244869566-31] StackTrace                               : Full Stack Trace:
rundeck_1                   |
rundeck_1                   | java.lang.NullPointerException: entry
rundeck_1                   |   at java.util.zip.ZipFile.getInputStream(ZipFile.java:361)
rundeck_1                   |   at java_util_zip_ZipFile$getInputStream$1.call(Unknown Source)
rundeck_1                   |   at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
rundeck_1                   |   at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
rundeck_1                   |   at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:128)
rundeck_1                   |   at com.rundeck.repository.client.util.ArtifactUtils.extractArtifactMetaFromZip(ArtifactUtils.groovy:91)
rundeck_1                   |   at com.rundeck.repository.client.util.ArtifactUtils$extractArtifactMetaFromZip$3.callStatic(Unknown Source)
rundeck_1                   |   at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallStatic(CallSiteArray.java:55)
rundeck_1                   |   at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:197)
rundeck_1                   |   at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:209)
rundeck_1                   |   at com.rundeck.repository.client.util.ArtifactUtils.getMetaFromUploadedFile(ArtifactUtils.groovy:73)
rundeck_1                   |   at com.rundeck.repository.client.util.ArtifactUtils$getMetaFromUploadedFile$1.callStatic(Unknown Source)
rundeck_1                   |   at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallStatic(CallSiteArray.java:55)
rundeck_1                   |   at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:197)
rundeck_1                   |   at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:209)
rundeck_1                   |   at com.rundeck.repository.client.util.ArtifactUtils.constructArtifactFileset(ArtifactUtils.groovy:53)
rundeck_1                   |   at com.rundeck.repository.client.util.ArtifactUtils$constructArtifactFileset$0.call(Unknown Source)
rundeck_1                   |   at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
rundeck_1                   |   at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
rundeck_1                   |   at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:128)
rundeck_1                   |   at com.rundeck.repository.client.repository.StorageTreeArtifactRepository.uploadArtifact(StorageTreeArtifactRepository.groovy:93)
rundeck_1                   |   at com.rundeck.repository.api.ArtifactRepository$uploadArtifact.call(Unknown Source)
rundeck_1                   |   at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
rundeck_1                   |   at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
rundeck_1                   |   at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:128)
rundeck_1                   |   at com.rundeck.repository.client.repository.RundeckRepositoryManager.uploadArtifact(RundeckRepositoryManager.groovy:127)
rundeck_1                   |   at com.rundeck.repository.api.RepositoryManager$uploadArtifact$1.call(Unknown Source)
rundeck_1                   |   at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
rundeck_1                   |   at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
rundeck_1                   |   at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:136)
rundeck_1                   |   at com.rundeck.repository.client.RundeckRepositoryClient.uploadArtifact(RundeckRepositoryClient.groovy:40)
rundeck_1                   |   at com.rundeck.repository.client.RepositoryClient$uploadArtifact$1.call(Unknown Source)
rundeck_1                   |   at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
rundeck_1                   |   at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
rundeck_1                   |   at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:136)
rundeck_1                   |   at repository.RepositoryController.uploadArtifact(RepositoryController.groovy:118)
rundeck_1                   |   at org.grails.core.DefaultGrailsControllerClass$MethodHandleInvoker.invoke(DefaultGrailsControllerClass.java:223)
rundeck_1                   |   at org.grails.core.DefaultGrailsControllerClass.invoke(DefaultGrailsControllerClass.java:188)
rundeck_1                   |   at org.grails.web.mapping.mvc.UrlMappingsInfoHandlerAdapter.handle(UrlMappingsInfoHandlerAdapter.groovy:90)
rundeck_1                   |   at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:967)
rundeck_1                   |   at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:901)
rundeck_1                   |   at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970)
rundeck_1                   |   at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:872)
rundeck_1                   |   at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
rundeck_1                   |   at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846)
rundeck_1                   |   at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
rundeck_1                   |   at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:865)
rundeck_1                   |   at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1655)
rundeck_1                   |   at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:215)
rundeck_1                   |   at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
rundeck_1                   |   at org.springframework.security.web.jaasapi.JaasApiIntegrationFilter$1.run(JaasApiIntegrationFilter.java:96)
rundeck_1                   |   at java.security.AccessController.doPrivileged(Native Method)
rundeck_1                   |   at javax.security.auth.Subject.doAs(Subject.java:422)
rundeck_1                   |   at org.springframework.security.web.jaasapi.JaasApiIntegrationFilter.doFilter(JaasApiIntegrationFilter.java:105)
rundeck_1                   |   at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
rundeck_1                   |   at org.springframework.boot.web.filter.ApplicationContextHeaderFilter.doFilterInternal(ApplicationContextHeaderFilter.java:55)
rundeck_1                   |   at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
rundeck_1                   |   at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
rundeck_1                   |   at org.springframework.boot.actuate.trace.WebRequestTraceFilter.doFilterInternal(WebRequestTraceFilter.java:111)
rundeck_1                   |   at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
rundeck_1                   |   at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
rundeck_1                   |   at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:317)
rundeck_1                   |   at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:127)
rundeck_1                   |   at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:91)
rundeck_1                   |   at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
rundeck_1                   |   at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:114)
rundeck_1                   |   at grails.plugin.springsecurity.web.UpdateRequestContextHolderExceptionTranslationFilter.doFilter(UpdateRequestContextHolderExceptionTranslationFilter.groovy:64)
rundeck_1                   |   at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
rundeck_1                   |   at grails.plugin.springsecurity.web.filter.GrailsAnonymousAuthenticationFilter.doFilter(GrailsAnonymousAuthenticationFilter.groovy:54)
rundeck_1                   |   at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
rundeck_1                   |   at org.springframework.security.web.authentication.rememberme.RememberMeAuthenticationFilter.doFilter(RememberMeAuthenticationFilter.java:158)
rundeck_1                   |   at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
rundeck_1                   |   at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:170)
rundeck_1                   |   at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
rundeck_1                   |   at org.springframework.security.web.jaasapi.JaasApiIntegrationFilter$1.run(JaasApiIntegrationFilter.java:96)
rundeck_1                   |   at java.security.AccessController.doPrivileged(Native Method)
rundeck_1                   |   at javax.security.auth.Subject.doAs(Subject.java:422)
rundeck_1                   |   at org.springframework.security.web.jaasapi.JaasApiIntegrationFilter.doFilter(JaasApiIntegrationFilter.java:105)
rundeck_1                   |   at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
rundeck_1                   |   at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:200)
rundeck_1                   |   at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
rundeck_1                   |   at grails.plugin.springsecurity.web.authentication.logout.MutableLogoutFilter.doFilter(MutableLogoutFilter.groovy:64)
rundeck_1                   |   at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
rundeck_1                   |   at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:105)
rundeck_1                   |   at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
rundeck_1                   |   at grails.plugin.springsecurity.web.SecurityRequestHolderFilter.doFilter(SecurityRequestHolderFilter.groovy:58)
rundeck_1                   |   at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
rundeck_1                   |   at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:214)
rundeck_1                   |   at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:177)
rundeck_1                   |   at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
rundeck_1                   |   at org.grails.web.servlet.mvc.GrailsWebRequestFilter.doFilterInternal(GrailsWebRequestFilter.java:77)
rundeck_1                   |   at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
rundeck_1                   |   at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
rundeck_1                   |   at org.grails.web.filters.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:67)
rundeck_1                   |   at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
rundeck_1                   |   at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
rundeck_1                   |   at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:197)
rundeck_1                   |   at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
rundeck_1                   |   at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
rundeck_1                   |   at org.springframework.web.filter.CorsFilter.doFilterInternal(CorsFilter.java:96)
rundeck_1                   |   at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
rundeck_1                   |   at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
rundeck_1                   |   at org.springframework.boot.actuate.autoconfigure.MetricsFilter.doFilterInternal(MetricsFilter.java:103)
rundeck_1                   |   at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
rundeck_1                   |   at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
rundeck_1                   |   at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)
rundeck_1                   |   at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)
rundeck_1                   |   at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)
rundeck_1                   |   at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
rundeck_1                   |   at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)
rundeck_1                   |   at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
rundeck_1                   |   at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
rundeck_1                   |   at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1317)
rundeck_1                   |   at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)
rundeck_1                   |   at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)
rundeck_1                   |   at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)
rundeck_1                   |   at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)
rundeck_1                   |   at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1219)
rundeck_1                   |   at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
rundeck_1                   |   at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
rundeck_1                   |   at org.eclipse.jetty.server.Server.handle(Server.java:531)
rundeck_1                   |   at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:352)
rundeck_1                   |   at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260)
rundeck_1                   |   at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:281)
rundeck_1                   |   at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:102)
rundeck_1                   |   at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)
rundeck_1                   |   at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
rundeck_1                   |   at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
rundeck_1                   |   at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
rundeck_1                   |   at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)
rundeck_1                   |   at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)
rundeck_1                   |   at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:762)
rundeck_1                   |   at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:680)
rundeck_1                   |   at java.lang.Thread.run(Thread.java:748)
```

**Comment by sjrd218** [comment]
@clofresh Are you uploading the jar file that is built after the plugin bootstrap generates the java project?

Generally the process that should work for jar files is the following:

1. Generate a java plugin using the plugin bootstrap
2. cd into the root dir created for your plugin and run `gradle build`
3. upload the jar that is created in the $plugin_root/builds/lib/your_plugin_0.1.jar to the repository with the rd client

**Comment by clofresh** [comment]
No, I'm bootstrapping a script plugin, so it's just a zip of the plugin.yaml and contents directory.

**Comment by sjrd218** [comment]
@clofresh Good catch. Looks like this happens because the pluginZip task puts a META-INF/MANIFEST.MF file in the archive, which causes it to be detected as a java plugin instead of a script plugin. I'll see if we can improve the detection code to make sure script plugins don't get incorrectly detected if they contain that file.

**Comment by sjrd218** [comment]
Fixed by c31bf8247ea9a7d15ba7bbd75cccb2b88d95fad1

**Comment by clofresh** [comment]
works now, thanks! :+1:
